src/packages/core/icon-registry/icon.registry.ts

```diff
@@ -13,26 +12,68 @@ interface UmbIconDescriptor {
  * @description - Icon Registry. Provides icons from the icon manifest. Icons are loaded on demand. All icons are prefixed with 'icon-'
  */
 export class UmbIconRegistry extends UUIIconRegistry {
+	#initResolve?: () => void;
+	#init: Promise<void> = new Promise((resolve) => {
+		this.#initResolve = resolve;
+	});
+
+	#icons: UmbIconDescriptor[] = [];
+	#unhandledRequests: { name: string; provider: UUIIconHost }[] = [];
+
+	setIcons(icons: UmbIconDescriptor[]) {
+		const oldIcons = this.#icons;
+		this.#icons = icons;
+		if (this.#initResolve) {
+			this.#initResolve();
+			this.#initResolve = undefined;
+		}
+		// Go through the new icons and see if they are already requested.
+		const newIcons = this.#icons.filter((i) => !oldIcons.find((o) => o.name === i.name));
+		newIcons.forEach((icon) => {
+			// Do we already have a request for this one
+			const unhandled = this.#unhandledRequests.find((i) => i.name === icon.name);
+			if (unhandled) {
+				if (unhandled.provider) {
+					this.#loadIcon(icon.name, unhandled.provider).then(() => {
+						this.#unhandledRequests.filter((i) => i.name !== icon.name);
```

Comment:
I dont believe that Array.prototype.filter replaces in-place. According to MDN, it produces a shallow copy of the elements that pass the test. I believe we need to do something like this:


```suggestion
						this.#unhandledRequests = this.#unhandledRequests.filter((i) => i.name !== icon.name);
```

Reply:
Oh yes, of course, thanks!